specs/protocol/holocene/exec-engine.md

```diff
@@ -73,5 +81,73 @@ an outbound withdrawal for a long period of time, the node may not have access t
 clients are able to at the very least reconstruct the account storage root at a given block on the fly if it does not
 directly store this information.
 
+## Extended `PayloadAttributesV3`
+
+The [`PayloadAttributesV3`](https://github.com/ethereum/execution-apis/blob/cea7eeb642052f4c2e03449dc48296def4aafc24/src/engine/cancun.md#payloadattributesv3)
+type is extended to:
+
+```rs
+PayloadAttributesV3: {
+    timestamp: QUANTITY
+    random: DATA (32 bytes)
+    suggestedFeeRecipient: DATA (20 bytes)
+    withdrawals: array of WithdrawalV1
+    parentBeaconBlockRoot: DATA (32 bytes)
+    transactions: array of DATA
+    noTxPool: bool
+    gasLimit: QUANTITY or null
+    eip1559Params: DATA (8 bytes)
+}
+```
+
+### `eip1559Params` encoding
+
+| Name          | Type  | Byte Offset |
+| ------------- | ----- | ----------- |
+| `denominator` | `u32` | `[0, 4)`    |
+| `elasticity`  | `u32` | `[4, 8)`    |
+
+### Execution
+
+During execution, the EIP-1559 parameters used to calculate the next block base fee should come from the
+`PayloadAttributesV3` type rather than the previous protocol constants, if it is non-null.
+
+- If, before Holocene activation, `eip1559Parameters` is non-zero, the attributes are to be considered invalid by the
+  engine.
+- After Holocene activation:
+  - if `eip1559Params` is zero, the [canyon base fee parameter constants](../exec-engine.md#1559-parameters) are
+    used.
+  - if `eip1559Params` are non-null, the values from the attributes are used.
+
+### Rationale
+
+This type is made available in the payload attributes to allow the block builder to dynamically control the EIP-1559
+parameters of the chain. As described in the [derivation - AttributesBuilder](./derivation.md#attributes-builder)
+section, the derivation pipeline must populate this field from the `SystemConfig` during payload building, similar to
+how it must reference the `SystemConfig` for `gasLimit` field.
+
+## `eip1559Params` in Header
+
+Upon Holocene activation, the L2 block header's `nonce` field will consist of the 8-byte `eip1559Params` value.
+
+### Header Validity Rules
+
+Prior to Holocene activation, the L2 block header's `nonce` field is valid iff it is equal to `u64(0)`.
+
+After Holocene activation, The L2 block header's `nonce` field is valid iff it is non-zero.
```

Comment:
Although the first Holocene block will still have the 0 value?

Reply:
The first holocene block will not have the `0` value - it will insert the canyon base fee params into the `nonce` field.

Reply:
oh gotcha!